examples/custom-widget-with-3rd-party-data/src/controller/custom-widget.ts

```diff
@@ -0,0 +1,47 @@
+import {
+    DDClient,
+    EventType,
+    WidgetSettingsMenuClickData
+} from '@datadog/ui-extensions-sdk';
+import * as React from 'react';
+import { logoutUser, User } from '../user';
+
+/**
+ * This hook performs any app-wide for the custom widget.
+ * @param client The initialized {@link DDClient}
+ */
+function useSetupCustomWidget(client: DDClient<User>): void {
+    /**
+     * We set up an event listener for the logout widget settings menu item.
+     * This event handler lets us perform the actual logging out of a user.
+     */
+    React.useEffect(() => {
+        const unsubscribeLogout = client.events.on(
+            EventType.WIDGET_SETTINGS_MENU_CLICK,
+            async (data: WidgetSettingsMenuClickData): Promise<void> => {
+                /**
+                 * We only want to handle events from the `'logout'` settings menu item.
+                 */
+                if (data.menuItem.key !== 'logout') {
+                    return;
+                }
+
+                /**
+                 * Perform the actual logout,
+                 * then make sure to update the auth state so it's reflected in Datadog.
+                 */
+                await logoutUser();
+                await client.auth.updateAuthState();
+            }
```

Comment:
I love it, it's never been clear how best to handle this. Devon suggested making 'logout' a first-class thing defined in the auth provider, but until then having this example is great